ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/runs/$runIndex/index.tsx

```diff
@@ -94,11 +117,15 @@ const RunComponent = () => {
     }
   };
 
-  async function handleDownload(runId: number, filename: string) {
-    await downloadZipFile({
-      runId: runId,
-      fileName: filename,
-    });
+  async function handleDownload(runId: number, filename?: string) {
+    if (!filename) {
+      await downloadLogs(runId);
+    } else {
+      await downloadReports({
+        runId: runId,
+        fileName: filename,
+      });
+    }
```

Comment:
I think using one function here adds some complexity without providing much value. I directly using `downloadLogs` and `downloadReports` in the `onClick`s should work without wrapping them in this function, or doesn't it?